modules/apps/29-fee/ibc_module.go

```diff
@@ -219,21 +219,9 @@ func (im IBCModule) OnAcknowledgementPacket(
 		return im.app.OnAcknowledgementPacket(ctx, packet, ack.Result, relayer)
 	}
 
-	// cache context before trying to distribute the fee
-	cacheCtx, writeFn := ctx.CacheContext()
+	im.keeper.DistributeFee(ctx, identifiedPacketFee.RefundAddress, ack.ForwardRelayerAddress, relayer.String(), packetId)
```

Comment:
modify `DistributeFee` API so you don't have to call `relayer.String()` and then parse it to sdk.AccAddress